lib/Middleware/SessionMiddleware.php

```diff
@@ -123,12 +123,14 @@ private function assertDocumentSession(ISessionAwareController $controller): voi
 	private function assertUserOrShareToken(ISessionAwareController $controller): void {
 		$documentId = (int)$this->request->getParam('documentId');
 		if (null !== $userId = $this->userSession->getUser()?->getUID()) {
-			// Check if user has access to document
-			if ($this->rootFolder->getUserFolder($userId)->getFirstNodeById($documentId) === null) {
-				throw new InvalidSessionException();
+			if (count($this->rootFolder->getUserFolder($userId)->getById($documentId)) !== 0) {
```

Comment:
Any reason to move away from `getFirstNodeById`? It would be the more performant approach and I'd prefer to keep that.

Reply:
fixed